nipoppy_cli/nipoppy/workflows/pipeline.py

```diff
@@ -124,19 +125,44 @@ def pipeline_config(self) -> ProcPipelineConfig:
     def fpath_container(self) -> Path:
         """Return the full path to the pipeline's container."""
         fpath_container = self.pipeline_config.get_fpath_container()
+
         if fpath_container is None:
             raise RuntimeError(
                 f"No container image file specified in config for pipeline"
                 f" {self.pipeline_name} {self.pipeline_version}"
             )
 
         elif not fpath_container.exists():
-            raise FileNotFoundError(
-                f"No container image file found at {fpath_container} for pipeline"
-                f" {self.pipeline_name} {self.pipeline_version}"
-            )
+
+            if not is_datalad_installed() or not self.is_datalad_dataset:
+                raise FileNotFoundError(
+                    f"No container image file found at {fpath_container} for pipeline"
+                    f" {self.pipeline_name} {self.pipeline_version}"
+                )
+
         return fpath_container
 
+    def _get_container_image(self):
```

Comment:
gonna be hard to test this one as the images to pull are too heavy for github CI